### ISSUE TITLE
x/tools/gopls: multiple failures on windows-arm-zx2c4

### ISSUE THREAD

**Issue by findleyr** [golang/go#44232]
Looking at our build dashboard for first-class ports, windows-arm-zx2c4 stands out as failing for a long time.

Some of the failures might be due to resource constraints (gopls tests can be very IO and memory heavy), but some are failures related to the filesystem that should be investigated, for example:
https://build.golang.org/log/477a5faf3f4383900706c4b25940a706322b6362

```
--- FAIL: TestLSP (23.64s)
    --- FAIL: TestLSP/Modules (8.57s)
        lsp_test.go:57: invalid workspace configuration: case mismatch in path "C:\\Users\\GOPHER~1.DES\\AppData\\Local\\Temp\\TestLSP_Modules233674203\\lsp": component "GOPHER~1.DES" should be "Gopher.DESKTOP-6IM5GCA"
        
--- FAIL: TestCaseInsensitiveFilesystem (0.02s)
    view_test.go:50: checkPathCase("C:\\Users\\GOPHER~1.DES\\AppData\\Local\\Temp\\TestCaseInsensitiveFilesystem086638055\\a\\B\\c\\DEFgh\\f.go") = case mismatch in path "C:\\Users\\GOPHER~1.DES\\AppData\\Local\\Temp\\TestCaseInsensitiveFilesystem086638055\\a\\B\\c\\DEFgh\\f.go": component "GOPHER~1.DES" should be "Gopher.DESKTOP-6IM5GCA", wanted error: false
```

Opening this issue to track getting this builder out of failure, either by deciding to skip these tests or by fixing them.

CC @stamblerre @heschik @pjweinbgo

**Comment by bcmills** [comment]
That builder is really slow in certain ways, so it has a habit of exposing tests that either require a lot of resources or make invalid assumptions about timing.

(See #42925, #43120, #43668.)

**Comment by findleyr** [comment]
Yeah, I suspect we'll need to skip tests that are resource intensive, but those path errors in particular might warrant consideration.

Also, as a point of clarification, this isn't a first class port (see #27689).

**Comment by bcmills** [comment]
The `case mismatch` failures are still present. It appears that the test is assuming that a particular component is referred to by its long name, but this builder is instead referring to it by its short name.

(Many Windows filesystems allow each path component to be referred to by both an 8.3 name and a longer name. Tests need to be redundant to those variations on Windows, in much the same way that they need to be redundant to `PWD` symlinks on macOS.)

**Comment by bcmills** [comment]
CC @zx2c4

**Comment by zx2c4** [comment]
I suspect this is a consequence of TMP/TEMP being set to something long. I'll try setting that to something shorter.

**Comment by zx2c4** [comment]
Alright, now it's set to `C:\temp`, instead of the thing with the 8.3 abbreviations in it. Let's see if that fixes some things.

**Comment by pjweinb** [comment]
This builder has been retired.
